app/src/main/java/com/hedvig/app/feature/loggedin/service/GetCrossSellsUseCase.kt

```diff
@@ -0,0 +1,13 @@
+package com.hedvig.app.feature.loggedin.service
+
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.CrossSellsQuery
+import com.hedvig.app.util.apollo.safeQuery
+
+class GetCrossSellsUseCase(
```

Comment:
I think it makes sense for this UseCase to handle the result of the query. Eg. moving this from `TabNotificationService`: 
```
val potentialCrossSells = when (val potentialCrossSellsResult = getCrossSellsUseCase.invoke()) {
            is QueryResult.Success -> {
                getCrossSells(potentialCrossSellsResult.data)
            }
            is QueryResult.Error -> {
                e { "Error when loading potential cross-sells: ${potentialCrossSellsResult.message}" }
                emptySet()
            }
        }

private fun getCrossSells(
        crossSellData: CrossSellsQuery.Data
    ) = crossSellData
        .activeContractBundles
        .flatMap { contractBundle ->
            contractBundle
                .potentialCrossSells
                .map { it.contractType.toString() }
        }
        .toSet()
```

Having some form of `CrossSellType` enum would also be nice so that we dont rely on network types outside of the network layer.

Reply:
Agreed.